src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.GenerateAssemblyInfo.targets

```diff
@@ -84,7 +81,35 @@ Copyright (c) .NET Foundation. All rights reserved.
         <_Parameter1>$(NeutralLanguage)</_Parameter1>
       </AssemblyAttribute>
     </ItemGroup>
+  </Target>
+
+  <!-- 
+    To allow version changes to be respected on incremental builds (e.g. through CLI parameters),
+    create a hash of all assembly attributes so that the lock file name will change with the calculated
+    assembly attribute values and msbuild will then execute CoreGenerateAssembly to generate a new file.
+  -->
+  <Target Name="_CreateGeneratedAssemblyInfoInputsCacheFile"
+          DependsOnTargets="_CalculateAssemblyAttributes"
+          BeforeTargets="CoreGenerateAssemblyInfo">
```

Comment:
> Does the DependsOnTargets="A;B;C" set the build dependency of "b depends on a"?

No, but it ensures A,B,C ordering when it triggers them to run. IIRC, there is precedent in Common targets to slot CoreXxx in via DependsOnTargets of Xxx order without explicitly saying that CoreXxx depends on everything before it in the DependsOnTargets of Xxx. That was the idiom I had in mind when I first wrote this, but I think you're on to something nicer. We should make sure that GetAssemblyVersion is also sequenced by the same mechanism:

``` XML
<Target Name="GenerateAssemblyInfo" 
        DependsOnTargets="PrepareForBuild;CoreGenerateAssemblyInfo"
        BeforeTargets="CoreCompile" ... />

<Target Name="GetAssemblyAttributes"
        DependsOnTargets="GetAssemblyVersion" .. />

<Target Name="_CreateGeneratedAssemblyInfoInputsCacheFile"
        DependsOnTargets="GetAssemblyAttributes"
        BeforeTargets="CoreGenerateAssemblyInfo"  ... />
```